### DISCUSSION TITLE
Set Power Limit via MQTT

### DISCUSSION THREAD

**Discussion by Mogdar-M** [hoylabs/OpenDTU-OnBattery#217] [Q&A] [answered]
Inverter is configured and producing.
MQTT is enabled and I can read data

But I can'T set a limit for the inventer via MQTT and IO Broker

With Ahoy it worked usinf this datapoint
mqtt.0.inverter.ctrl.limit.0

This datapoint doesn't exists ob openDTU on battery

I've tried to set the value on
mqtt.0.openDTU.1141xxxxxxx.status.limit_relative
but with no effect

So, I know how to set up a script for it but I'm missing the target on the inverter

**Comment by helgeerbe** [comment]
https://github.com/tbnobody/OpenDTU/blob/master/docs/MQTT_Topics.md

**Comment by Mogdar-M** [comment]
yes, tried it:
 setState("mqtt.0.openDTU.11418xxxxxx./cmd/limit_persistent_relative"/*openDTU/11418xxxxxx//cmd/limit_persistent_relative*/, 71);
 
 did not work

**Reply by helgeerbe** [reply]
This doesn't look right. If you didn't change the top topic. It should start with solar. Read top line of the link I sent.

**Reply by Mogdar-M** [reply]
I changed it as Solar was already in use
![image](https://user-images.githubusercontent.com/129499239/235652283-cceeec64-a017-4f5b-b68f-7d04548f6a8b.png)


And I'm receiving all data from DTU and victron via MQTT
![image](https://user-images.githubusercontent.com/129499239/235652083-7b74ae53-8d7e-4e89-8180-12877d5939ed.png)

But the folder for cmd/limit is not available so I can't update it

**Comment by Mogdar-M** [comment]
looks like a generic openDTU topic

**Comment by Mogdar-M** [comment, marked as answer]
just to share the solution:
manually create all entries for the cmd folder
using this parameter
https://github.com/tbnobody/OpenDTU/blob/master/docs/MQTT_Topics.md
install another mqtt instance in IO Broker as subscriber
reboot instance mqtt 0
instance mqtt 1 will generate duplicates of all mqtt 0 entries

use a similar script to set the limit via mqtt in mqtt 1
![image](https://user-images.githubusercontent.com/129499239/235889517-8000b477-b1c8-462f-8670-fcf24ca07c81.png)
